src/rgw/rgw_exporter.cc

```diff
@@ -0,0 +1,59 @@
+#include "rgw_exporter.h"
+
+#include <chrono>
+#include <utility>
+
+#include "common/perf_counters_key.h"
+
+using namespace std::chrono_literals;
+using namespace rgw::usage_counters;
+
+namespace rgw{
+
+RGWExporter::RGWExporter(CephContext* cct, RGWUsageCache* cache)
+  : cct(cct), cache(cache) {}
+
+RGWExporter::~RGWExporter() {
+  stop();
+}
+
+void RGWExporter::start() {
+  stop_requested = false;
+  worker = std::thread(&RGWExporter::run, this);
+}
+
+void RGWExporter::stop() {
+  stop_requested = true;
+  if (worker.joinable()) {
+    worker.join();
+  }
+}
+
+void RGWExporter::run() {
+  while (!stop_requested) {
+    refresh();
+    std::this_thread::sleep_for(1s);
+  }
+}
+
+void RGWExporter::refresh() {
+  /* Placeholder for reading usage from metadata and updating counters */
```

Comment:
This placeholder has me rather concerned. It looks like we don't actually gather the statistics from RGWRados. Did you forget to push the later commits?